pkg/upgrade/control_plane.go

```diff
@@ -24,6 +24,7 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	bootstrapv1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
 	kubeadmv1beta1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/kubeadm/v1beta1"
+	kubeadmv1beta2 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/kubeadm/v1beta2"
```

Comment:
Do you need v1beta2 specifically? It looks like the ClusterStatus type is the same in both versions?

Reply:
I saw v1beta2 in my configmap so just went for that. will it work either way, you think?

Reply:
cc @chuckha

Reply:
I'd stick with v1beta1 if we can

Reply:
so do we want to do some unstructured goodies here? or what do we think would be most appropriate?

also, so I'm understanding, is this just so we can avoid importing the extra package?

Reply:
Mostly future-proofing so we don't have to fix this in the future if the api version changes but the structure remains the same